src/core/histogram.cc

```diff
@@ -96,6 +104,65 @@ histogram::print(FILE *fp) const {
                                  CLASSIC);  // Format CLASSIC/CSV supported.
 }
 
+seastar::metrics::histogram
+histogram::get_seastar_metrics_histogram() const {
+  // logarithmic histogram configuration. this will range from 10 microseconds
+  // through around 6000 seconds with 26 buckets doubling.
+  //
+  // TODO:
+  //   1) expose these settings through arguments
+  //   2) upstream log_base fix for sub-2.0 values. in hdr the log_base is a
+  //   double but is truncated (see the int64_t casts on log_base below which is
+  //   the same as in the hdr C library). this means that if we want buckets
+  //   with a log base of 1.5, the histogram becomes linear...
+  const size_t num_buckets = 26;
+  const int64_t first_value = 10;
+  const double log_base = 2.0;
+
+  seastar::metrics::histogram sshist;
+  sshist.buckets.resize(num_buckets);
+
+  sshist.sample_count = hist_->sample_count;
+  sshist.sample_sum = (double)hist_->sample_sum;
+
+  // stack allocated; no cleanup needed
+  struct hdr_iter iter;
+  struct hdr_iter_log *log = &iter.specifics.log;
+  hdr_iter_log_init(&iter, hist_->hist, first_value, log_base);
+
+  // fill in buckets from hdr histogram logarithmic iteration. there may be more
+  // or less hdr buckets reported than what will be returned to seastar.
+  size_t bucket_idx = 0;
+  while (hdr_iter_next(&iter) && bucket_idx < sshist.buckets.size()) {
```

Comment:
why not `for(; hdr_iter_next(&iter) && bucket_idx < sshist.buckets.size(); ++bucket_idx)` so you don't forget to update the index. Either is ok.